### PR TITLE
Exosuit fabricator wires and hacking

### LINKED ISSUE #15305: adamantine free golems cant do research, also likely not the mech fab
### Severity

Inconvenience

### Short Description

started round as free golem and tried to research, got access denied, likely going to be the same with the mech  fab

### Reproduction

access the rnd console and click on a research tab

### Round ID

42329

### Test Merges

_No response_

### BYOND Username

_No response_

### Additional information

likely by the mech fab change

### LINKED ISSUE #15322: Free miners can't use exosuit fabricator.
### Severity

Inconvenience

### Short Description

Due to the new exosuit fabricator update, free miners were unintentionally removed from the ability to build borgs and exosuits.

### Reproduction

1. Be free miner
2. Build exo fabricator
3. Click on it
4. Invalid credentials

### Round ID

42341

### Test Merges

_No response_

### BYOND Username

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Majkl-J**
# This does not allow for making combat designs
The reason for this change is simple: Exofabs in their current state need a **roboticist id to even open and use**. This is usually an issue during lowpops (Forces you to greytide all the way for the spare), also an issue for multiple ghost roles.
It is also inconsistent with the rest of the potentially dangerous machines (Chemistry, basically every lathe, etc.), which don't have any id checks whilst being potentially more destructive (The only potentially harmful thing in the fab is flashes)

# Document the changes in your pull request
Adds wires to the exofab, 2 duds 3 functional:

Hack wire: When cut, allows you to open the exofab without an id. When pulsed, blinks a light.
Shock wire: When pulsed, shocks the exofab for a short time.
Zap wire: When pulsed, makes the manipulator hands spazz out and break one of the hacker's limbs.

close #15322 
close #15305

# Wiki Documentation
Needs to be added to the Hacking page with a small mention on the robotics/roboticist one.

# Changelog

:cl:  
rscadd: Added wires to the exosuit fabricator
tweak: The exosuit fabricator can now be hacked
/:cl:

### CODE REVIEW

**Comment by Sniblet** [comment]
So it's like exosuit fabricators were before the stupid nerf, except as a compromise we break the roboticist's arms every other round

**Comment by Maia-J** [comment]
> So it's like exosuit fabricators were before the stupid nerf, except as a compromise we break the roboticist's arms every other round

The roboticist doesn't need to hack it, they have the access. But yes, as a compromise we do break those arms here and there. 

If anything, you can yell at me if it feels too harsh/it happens too often once and if this gets merged. I can always balance it out once I get some more feedback.

**Comment by iamthesilly** [comment]
> > So it's like exosuit fabricators were before the stupid nerf, except as a compromise we break the roboticist's arms every other round
> 
> The roboticist doesn't need to hack it, they have the access. But yes, as a compromise we do break those arms here and there.
> 
> If anything, you can yell at me if it feels too harsh/it happens too often once and if this gets merged. I can always balance it out once I get some more feedback.

free miner engineers and other ghost roles will need to hack it just so they can do the things they are supposed to do while risking severe harm in doing so, and there isn't any reason for it to be ID locked anyway when all the dangerous stuff is locked behind command authorization or red alert

**Comment by Maia-J** [comment]
> free miner engineers and other ghost roles will need to hack it just so they can do the things they are supposed to do while risking severe harm in doing so, and there isn't any reason for it to be ID locked anyway when all the dangerous stuff is locked behind command authorization or red alert

Yea but when trying to downright remove the id lock Jamie said no so this is in my opinion the best option. Also both Charlie and Free miners have access to a medbay with tools by the time they reach their exofabs. And golems can't get fractures far as I know.

**Comment by iamthesilly** [comment]
> Yea but when trying to downright remove the id lock Jamie said no so this is in my opinion the best option. Also both Charlie and Free miners have access to a medbay with tools by the time they reach their exofabs. And golems can't get fractures far as I know. 

Did Jamie say no to only having it shock you instead of breaking your limbs? Also if it's going to break a limb it should the the arm that's holding the tool you're using, but that's just my take.

**Comment by Maia-J** [comment]
> Did Jamie say no to only having it shock you instead of breaking your limbs? Also if it's going to break a limb it should the the arm that's holding the tool you're using, but that's just my take.

Actually, I could do that only tool arm thing. As for it having only shocking, I wanted to switch hacking up for a bit so its not just shocking on everything. Its a complicated machine and having it shock you just like everything else feels kinda dull.

**Comment by iamthesilly** [comment]
> Actually, I could do that only tool arm thing. As for it having only shocking, I wanted to switch hacking up for a bit so its not just shocking on everything. Its a complicated machine and having it shock you just like everything else feels kinda dull.

Honestly that's a good reason, though it would be nice to have something show up in the chat so you know why your arm is suddenly broken

**Comment by Maia-J** [comment]
> > Actually, I could do that only tool arm thing. As for it having only shocking, I wanted to switch hacking up for a bit so its not just shocking on everything. Its a complicated machine and having it shock you just like everything else feels kinda dull.
> 
> Honestly that's a good reason, though it would be nice to have something show up in the chat so you know why your arm is suddenly broken

There is text about it, but come to think of it, it's not easy to notice so I will adjust that. I will leave the limb breaking as is currently and wait for more feedback about that.

Thanks for the feedback!

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]: Don’t create a new body part?

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: Ok on it, this should work much better
```
	var/picked_bodypart = pick(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG)
	var/obj/item/bodypart/bone = C.get_bodypart(picked_bodypart)
```

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]: dont call it zap then

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]: no braces

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm:178]: delete the wound if you don't apply it

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm:15]: doesnt do anything

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm:15]: Fixed. Also var/hacked is used at L 557

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: Was like that for consistent naming with the wire as defining a new one would be a bit too needless. Changed anyway

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm:178]: Ok I won't resolve this one as I have no clue how Byond memory management works. I just used qdel() and hope it's correct

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: Ok can I please get some elaboration on this? I have optimised it to
`var/obj/item/bodypart/bone = C.get_bodypart(pick(BODY_ZONE_R_ARM, BODY_ZONE_L_ARM, BODY_ZONE_R_LEG, BODY_ZONE_L_LEG))`
to lower the memory impact but it still isnt in any sort of brackets itself.

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm:178]: Nvm got it

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: Ah I see I am blind

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]: dont delete their bodypart?

File: code/game/mecha/mech_fabricator.dm
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]: only delete the wound if you don't use it

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: Woopsie one less indent

File: code/game/mecha/mech_fabricator.dm
Comment (Maia-J) [reply code/game/mecha/mech_fabricator.dm]: There. fixed and done hopefully

**Comment by Maia-J** [comment]
I can now say with 80% confidence that the exofab will function as intended now. After testing it, all works as this PR's description states.

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**
